freetextresponse/freetextresponse.py

```diff
@@ -494,13 +525,44 @@ def submit(self, data, suffix=''):
             'indicator_class': self._get_indicator_class(),
             'used_attempts_feedback': self._get_used_attempts_feedback(),
             'submit_class': self._get_submit_class(),
-            'word_count_message': self._get_word_count_message(
-                ignore_attempts=True
-            ),
             'submitted_message': self._get_submitted_message(),
+            'save_class': self._get_submit_class(),
+            'user_alert': self._get_user_alert(
+                ignore_attempts=True,
+            ),
+            'indicator_visibility_class': self._get_indicator_visiblity_class(),
         }
         return result
 
+    @XBlock.json_handler
+    def save_reponse(self, data, suffix=''):
+        # pylint: disable=unused-argument
+        """
+        Processes the user's save
+        """
+        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
+            raise StandardError(
```

Comment:
Doesn't this throw a 500? I know we're already doing this in the submit function but this isn't a server error. Can we just use the logger in both places?